src/usePerpsMetadata.ts

```diff
@@ -0,0 +1,27 @@
+import { useQuery } from '@tanstack/react-query';
+import { ethers } from 'ethers';
+import { useImportContract } from './useImports';
+import { useSynthetix } from './useSynthetix';
+
+interface PerpsMetadata {
+  name: string;
+  symbol: string;
+}
+
+export function usePerpsMetadata({ provider, marketId }: { provider?: ethers.providers.BaseProvider; marketId?: ethers.BigNumber }) {
+  const { chainId } = useSynthetix();
+  const { data: PerpsMarketProxyContract } = useImportContract('PerpsMarketProxy');
+
+  return useQuery<PerpsMetadata>({
+    enabled: Boolean(chainId && provider && marketId && PerpsMarketProxyContract?.address),
+    queryKey: [chainId, 'Perps Metadata', { PerpsMarketProxy: PerpsMarketProxyContract?.address }, { marketId: marketId?.toString() }],
+    queryFn: async () => {
+      if (!(chainId && provider && marketId && PerpsMarketProxyContract?.address)) {
+        throw 'OMFG';
+      }
+
+      const PerpsMarketProxy = new ethers.Contract(PerpsMarketProxyContract.address, PerpsMarketProxyContract.abi, provider);
+      return PerpsMarketProxy.metadata(marketId);
```

Comment:
Even better to be more explicit in return values so we dont need to be as specific in types...

```suggestion
      const {symbol, name} = PerpsMarketProxy.metadata(marketId);
      return {
        symbol, 
        name
      }
```

Reply:
add `return { symbol, name }`